src/main/kotlin/net/irisshaders/lilybot/extensions/util/RoleMenu.kt

```diff
@@ -0,0 +1,190 @@
+package net.irisshaders.lilybot.extensions.util
+
+import com.kotlindiscord.kord.extensions.DISCORD_BLACK
+import com.kotlindiscord.kord.extensions.commands.Arguments
+import com.kotlindiscord.kord.extensions.commands.converters.impl.defaultingColor
+import com.kotlindiscord.kord.extensions.commands.converters.impl.defaultingString
+import com.kotlindiscord.kord.extensions.commands.converters.impl.optionalChannel
+import com.kotlindiscord.kord.extensions.commands.converters.impl.role
+import com.kotlindiscord.kord.extensions.components.components
+import com.kotlindiscord.kord.extensions.components.ephemeralButton
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import com.kotlindiscord.kord.extensions.extensions.ephemeralSlashCommand
+import com.kotlindiscord.kord.extensions.extensions.event
+import com.kotlindiscord.kord.extensions.types.respond
+import dev.kord.common.entity.ButtonStyle
+import dev.kord.core.behavior.channel.createMessage
+import dev.kord.rest.builder.message.create.embed
+import net.irisshaders.lilybot.utils.FULLMODERATORS
+import com.kotlindiscord.kord.extensions.utils.hasRole
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.behavior.channel.GuildMessageChannelBehavior
+import dev.kord.core.behavior.channel.createEmbed
+import dev.kord.core.behavior.interaction.respondEphemeral
+import dev.kord.core.entity.interaction.ButtonInteraction
+import dev.kord.core.event.interaction.InteractionCreateEvent
+import kotlinx.datetime.Clock
+import net.irisshaders.lilybot.database.DatabaseManager
+import net.irisshaders.lilybot.utils.MOD_ACTION_LOG
+import org.jetbrains.exposed.sql.insertIgnore
+import org.jetbrains.exposed.sql.select
+import org.jetbrains.exposed.sql.transactions.experimental.newSuspendedTransaction
+
+class RoleMenu : Extension() {
+    override val name = "rolemenu"
+
+	override suspend fun setup() {
+		ephemeralSlashCommand(::RoleMenuArgs) {
+			name = "role-menu"
+			description = "Creates a menu that allows users to select a role."
+
+			allowRole(FULLMODERATORS)
```

Comment:
Can you change this to line up with the new system on develop please. I can handle the rest once merged

Reply:
Done.